external-crates/move/crates/move-vm-profiler/src/lib.rs

```diff
@@ -184,18 +179,29 @@ impl GasProfiler {
     }
 
     pub fn to_file(&self) {
-        if !*PROFILER_ENABLED || !self.is_metered() {
+        let config;
+        if let Some(c) = &self.config {
+            config = c
+        } else {
+            return;
+        }
```

Comment:
```suggestion
        let Some(config) = &self.config else {
            return;
        };
```

Reply:
sadly this will not capture the config for use below

Reply:
Hmm, why not? That's the whole point of `let ... else` -- what's the error you get when you try this?